common/include/pcl/point_cloud.h

```diff
@@ -535,6 +612,28 @@ namespace pcl
         height = 1;
       }
 
+      /**
+       * \brief Replaces the points with the elements from the initializer list `ilist`
+       * \note This calculates the height based on size and width provided. This means
+       * the assignment happens even if the size is not perfectly divisible by width
+       * \param[in] new_width new width of the point cloud
+       */
+      void
+      assign(std::initializer_list<PointT> ilist, index_t new_width)
```

Comment:
Here we could also add a new parameter with default value to the function above:
`assign(std::initalizer_list<PointT> ilist, uindex_t new_height=0)`